frontend/src/libs/components/tooltip/element-with-tooltip.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  createElement,
+  type FC,
+  type ReactElement,
+  type ReactHTML,
+  type ReactNode,
+} from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+import { DataTooltipId } from '~/libs/enums/enums.js';
+import { type ValueOf } from '~/libs/types/types.js';
+
+type ElementWithTooltipProperties = {
+  className?: string;
+  elementType: keyof ReactHTML;
+  placement?: 'bottom' | 'top' | 'left' | 'right';
```

Comment:
Maybe, it will be better to move it to enum?

Reply:
I believe we won't use this enum any way except for type here, so both approaches applicable - as enum or as union type

Reply:
@andanf-e  Should I move this tunion type to a separate file, or can I leave it as is? Because it's not clear from the context!

Reply:
If we'd use union - then it may be left here

Reply:
[70ef25d](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/156/commits/70ef25d08fe34060a1c27efcc6bec2510a5e544f)